provisioning/atat_provisioning_api.yaml

```diff
@@ -0,0 +1,524 @@
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PortfolioDraftSummary'
+      requestBody:
+        description: An empty String or JSON body
+        content:
+          application/json:
+            schema:
+              type: string
+              example: '{}'
+          text/plain:
+            schema:
+              type: string
+              example: ''
+  '/portfolioDrafts/{portfolioDraftId}':
+    get:
+      tags:
+        - portfolioDraft
+      description: Gets a Portfolio Draft
+      operationId: getPortfolioDraft
+      parameters:
+        - name: portfolioDraftId
+          in: path
+          description: ID of the Portfolio
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PortfolioDraft'
+        '404':
+          description: Portfiolio Draft with the given ID does not exist
+    delete:
+      tags:
+        - portfolioDraft
+      description: Deletes a Portfolio Draft
+      operationId: deletePortfolioDraft
+      parameters:
+        - name: portfolioDraftId
+          in: path
+          description: ID of the Portfolio
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+        '404':
+          description: Portfiolio Draft with the given ID does not exist
+  '/portfolioDrafts/{portfolioDraftId}/portfolio':
+    get:
+      tags:
+        - portfolioDraft
+      description: Gets the Portfolio Step of the Portfolio Draft Wizard
+      operationId: getPortfolioStep
+      parameters:
+        - name: portfolioDraftId
+          in: path
+          description: ID of the Portfolio
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation with empty response body
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PortfolioStep'
+        '404':
+          description: Portfolio step not found for this Portfolio Draft
+        '504':
+          description: The given Portfolio Draft does not exist
+    post:
+      tags:
+        - portfolioDraft
+      description: Submits the Portfolio Step of the Portfolio Draft Wizard
+      operationId: createPortfolioStep
+      parameters:
+        - name: portfolioDraftId
+          in: path
+          description: ID of the Portfolio
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation with empty response body
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PortfolioStep'
+        '405':
```

Comment:
Is 405 the right status code? We allow POSTs but the client failed to make a valid request. Would 400 be more appropriate? Any scenario where a 405 feels appropriate would be handled by the API management service.

Reply:
Yeah that's probably fair. Will update.